micro_dl/utils/preprocess_utils.py

```diff
@@ -0,0 +1,95 @@
+    mask name and the corresponding file name. Each file_name should exist in
+    input_dir and belong to the same channel.
+    This function checks that all file names exist in input_dir and writes a
+    frames_meta csv containing mask names the indices corresponding to the
+    matched file_name. It also assigns a mask channel number for future
+    preprocessing steps like tiling.
+
+    :param dict pp_config: Preprocessing config
+    :return int mask_channel: New channel index for masks for writing tiles
+    :raises AssertionError: If 'masks' in pp_config contains both channels
+        and mask_dir (the former is for generating masks from a channel)
+    :raises IOError: If no csv file is present in mask_dir
+    :raises AssertionError: If more than one csv file exists in mask_dir
+        and no csv_name is provided to resolve ambiguity
+    :raises AssertionError: If provided csv_name if 'frames_meta.csv' - that
+        name is reserved for preprocessing output
+    :raises AssertionError: If csv doesn't consist of two columns named
+        'mask_name' and 'file_name'
+    :raises IndexError: If unable to match file_name in mask_dir csv with
+        file_name in input_dir for any given mask row
+    :raises AssertionError: If mask file correspond to more than one input
+        channel
+    """
+    # Masks need to have their own channel index for tiling
+    # Hopefully this will be big enough default value
+    mask_channel = 999
+    assert 'channels' not in pp_config['masks'], \
+        "Don't specify channels to mask if using pre-generated masks"
+    mask_dir = pp_config['masks']['mask_dir']
+    # Look for a csv. If more than one, get name from config
+    csv_name = glob.glob(os.path.join(mask_dir, '*.csv'))
+    if len(csv_name) == 0:
+        raise IOError("No csv file present in mask dir")
+    else:
+        # See if frames_meta is already present, if so, move on
+        has_meta = next((s for s in csv_name if 'frames_meta.csv' in s), None)
+        if isinstance(has_meta, str):
+            return mask_channel
+        if len(csv_name) == 1:
+            # Use the one existing csv name
+            csv_name = csv_name[0]
+        else:
+            # If more than one csv, we need to know which one
+            assert 'csv_name' in pp_config['masks'], \
+                "Please add csv_name to config->masks to resolve ambiguity"
+            csv_name = pp_config['masks']['csv_name']
+
+    # Read csv with masks and corresponding input file names
+    mask_meta = aux_utils.read_meta(input_dir=mask_dir, meta_fname=csv_name)
+
+    assert len(set(mask_meta).difference({'file_name', 'mask_name'})) == 0,\
+        "mask csv should have columns mask_name and file_name " +\
+        "(corresponding to the file_name in input_dir)"
+    # Check that file_name for each mask_name matches files in input_dir
+    input_dir = pp_config['input_dir']
+    input_meta = aux_utils.read_meta(input_dir)
```

Comment:
would be this be the output of generate_meta.py (a df with filename + channel, time, pos and z ids)?

Reply:
Yes. The function just adds some assertions and checks that there exists an input image for each mask, the output is the same.